frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx

```diff
@@ -128,34 +207,32 @@ function FieldValuesWidgetInner({
     }
   });
 
-  const _cancel = useRef(null);
+  const _cancel = useRef<null | (() => void)>(null);
 
   useUnmount(() => {
-    if (_cancel.current) {
-      _cancel.current();
-    }
+    _cancel?.current?.();
   });
 
-  const fetchValues = async query => {
+  const fetchValues = async (query?: string) => {
     setLoadingState("LOADING");
     setOptions([]);
 
-    let newOptions = [];
+    let newOptions: FieldValue[] = [];
     let newValuesMode = valuesMode;
     try {
-      if (canUseDashboardEndpoints(dashboard)) {
+      if (dashboard && canUseDashboardEndpoints(dashboard)) {
         const { values, has_more_values } = await fetchDashboardParameterValues(
           query,
         );
         newOptions = values;
         newValuesMode = has_more_values ? "search" : newValuesMode;
-      } else if (canUseCardEndpoints(question)) {
+      } else if (question && canUseCardEndpoints(question)) {
         const { values, has_more_values } = await fetchCardParameterValues(
           query,
         );
         newOptions = values;
         newValuesMode = has_more_values ? "search" : newValuesMode;
-      } else if (canUseParameterEndpoints(parameter)) {
+      } else if (parameter && canUseParameterEndpoints(parameter)) {
```

Comment:
`canUseParameterEndpoints` checks that non-nullable `parameter` is not null but it will never be called because of `parameter &&`

`canUseCardEndpoints` and `canUseDashboardEndpoints` also assume that the value passes there can be null so they have the optional chaining operator. Wdyt about making these 3 functions accept null and remove `quesiton &&` and similar checks?